tasks/check-data-keys/check-data-keys.yaml

```diff
@@ -4,17 +4,22 @@ kind: Task
 metadata:
   name: check-data-keys
   labels:
-    app.kubernetes.io/version: "0.9.2"
+    app.kubernetes.io/version: "0.10.0"
   annotations:
     tekton.dev/pipelines.minVersion: "0.12.1"
     tekton.dev/tags: release
 spec:
   description: >-
-    Tekton task to check that all required information is present to use the specified system(s)
+    Tekton task to check that all required information is present to use the specified system(s) 
+    and the data is valid against the schema.
   params:
     - name: dataPath
       description: Path to the JSON string of the merged data to use
       type: string
+    - name: schema 
+      description: URL to the JSON schema file to validate the data against
+      type: string
+      default: https://raw.githubusercontent.com/konflux-ci/release-service-catalog/refs/heads/production/schema/dataKeys.json
```

Comment:
I think I am okay with just having the default here like you have it and not passing it in the pipeline definitions. I don't think we can easily make it work for ci anyways

Reply:
Ok, so I have removed the schema parm in the pipelines

Reply:
That's fair based on what my comments were, but didn't we talk about in the call to have the value use taskGitRevision value? So then it would have to be passed in the pipelines

Reply:
Yes, I can change it.  I am just scared it will be harder to pass to e2e ....

Reply:
But isn't the update to the schema just adding the required keys? So the merged one (that e2e will run with) has nothing required, so it will pass?

Reply:
So e2e has an extra key [here](https://github.com/konflux-ci/e2e-tests/blob/c69d596d294d58aee5aa270d38783cc7d613b5f7/tests/release/pipelines/multiarch_advisories.go#L238) then the product_id is defined as int in the schema and in the [docs](https://konflux-ci.dev/release-service-docs/reference/catalog/data-key-options/#:~:text=source%3A%20%3Cstring%3E-,product_id,-%3A%20%3Cinteger%3E) but in e2e its a [string](https://github.com/konflux-ci/e2e-tests/blob/c69d596d294d58aee5aa270d38783cc7d613b5f7/tests/release/pipelines/multiarch_advisories.go#L252) so it will fail for time being need update the schema to include the source, git & url

Reply:
https://github.com/konflux-ci/e2e-tests/blob/c69d596d294d58aee5aa270d38783cc7d613b5f7/tests/release/pipelines/multiarch_advisories.go#L238 is wrong and should be fixed
product_id I'd have to check some users to see what the real thing is. e2e could be wrong for that too

Reply:
Ok, perfect then it should pass e2e, then 🙏